runtime/vm/ownedmonitors.c

```diff
@@ -149,6 +150,13 @@ walkFrameMonitorEnterRecords(J9VMThread *currentThread, J9StackWalkState *walkSt
 	IDATA monitorCount = (IDATA)walkState->userData2;
 	J9VMThread *targetThread = walkState->walkThread;
 
+	/* check if hidden method frame should be skipped */
+	if (J9_ARE_NO_BITS_SET(walkState->javaVM->runtimeFlags, J9_RUNTIME_SHOW_HIDDEN_FRAMES) && J9_IS_HIDDEN_METHOD(walkState->method)) {
```

Comment:
Does this also skip the methods inlined into a hidden frame?

Reply:
inlined methods should be walked by `jitGetOwnedObjectMonitors` in `jswalk.c` which check each inlined method inside `inlineMap`

Reply:
I agree the code looks like it flows the JIT frames `jitGetOwnedObjectMonitors` but the `isObjectStackAllocated` checks within this method leave me questioning if it is possible for this to be called on a jitted frame.  Otherwise, how'd we end up with a stack allocated object in a non-jitted frame?  peeking?

Reply:
Good point, maybe this check is for native frames? @gacholio do you have any insight

Reply:
A stack allocated object can appear in an interpreter frame. This can occur when the JIT does peeking

Reply:
When cannot report any stack allocated objects as this circumvents the anaylis the JIT does for these objects and potentially exposes a stale reference to tooling agents

Reply:
Thanks for confirming Tobi.  Approving now